query/src/main/scala/filodb/query/exec/RemoteExec.scala

```diff
@@ -24,8 +27,99 @@ import filodb.core.query.{PromQlQueryParams, QuerySession, QueryStats, QueryWarn
 import filodb.core.store.ChunkSource
 import filodb.query._
 
+
 trait RemoteExec extends LeafExecPlan with StrictLogging {
 
+  // Executes tasks immediately on the current thread.
+  val inProcessScheduler = Scheduler(new ExecutorService {
+      var _shutdown = false
+      def getThrowable[T](future: util.concurrent.Future[T]): Option[Throwable] = {
+        try {
+          future.get()
+        } catch {
+          case t: Throwable => return Some(t)
+        }
+        None
+      }
+      override def shutdown(): Unit = _shutdown = true
+      override def shutdownNow(): util.List[Runnable] = util.List.of()
+      override def isShutdown: Boolean = _shutdown
+      override def isTerminated: Boolean = _shutdown
+      override def awaitTermination(timeout: Long, unit: TimeUnit): Boolean = {
+        val waitUntilMs = System.currentTimeMillis() + unit.toMillis(timeout)
+        // looping in case thread is woken up early
+        while (!_shutdown && System.currentTimeMillis() < waitUntilMs) {
+          val waitMs = waitUntilMs - System.currentTimeMillis()
```

Comment:
This can possibly become negative between your checks and calculation and will end up with ``IllegalArgumentException``

Reply:
Good catch -- I assumed wrongly that negative values were ignored. Updated ✅